src/Microsoft.TestPlatform.Extensions.TrxLogger/Utility/FileHelper.cs

```diff
@@ -133,6 +133,45 @@ public static string GetNextIterationFileName(string parentDirectoryName, string
             return GetNextIterationNameHelper(parentDirectoryName, originalFileName, new FileIterationHelper(checkMatchingDirectory));
         }
 
+        /// <summary>
+        /// Constructs and returns first available timestamped file name. 
+        /// This does not checks for the file permissions.
+        /// </summary>
+        /// <param name="directoryName">Directory to try timestamped file names in.</param>
+        /// <param name="fileName">Filename (with extension) of the desired file. Timestamp will be added just before extension.</param>
+        /// <param name="timestampFormat">Timestamp format to be passed into DateTime.ToString method.</param>
+        /// <returns>First available filename with the format of `FileName{Timestamp}.ext`.</returns>
+        /// <example>
+        ///     <code>GetNextTimestampFileName("c:\data", "log.txt", "_yyyyMMddHHmmss")</code> will return "c:\data\log_20200801185521.txt", if available.
+        /// </example>
+        public static string GetNextTimestampFileName(string directoryName, string fileName, string timestampFormat)
+        {
+            EqtAssert.StringNotNullOrEmpty(directoryName, "parentDirectoryName");
+            EqtAssert.StringNotNullOrEmpty(fileName, "fileName");
+            EqtAssert.StringNotNullOrEmpty(timestampFormat, "timestampFormat");
+
+            uint iteration = 0;
+            var iterationStamp = DateTime.Now;
+            var fileNamePrefix = Path.GetFileNameWithoutExtension(fileName);
+            var extension = Path.GetExtension(fileName);
+            do
+            {
+                var tryMe = fileNamePrefix + iterationStamp.ToString(timestampFormat, DateTimeFormatInfo.InvariantInfo) + extension;
+
+                string tryMePath = Path.Combine(directoryName, tryMe);
+                if (!File.Exists(tryMePath))
+                {
+                    return tryMePath;
+                }
+
+                iterationStamp = iterationStamp.AddSeconds(1);
+                ++iteration;
+            }
+            while (iteration <= ushort.MaxValue);
```

Comment:
Why `iteration` has `uint` type, but in while loop you are comparing it with `ushort.MaxValue`?

Reply:
`uint.MaxValue` is too high. So I used `ushort` there.